src/ui/VirtualRenderers/VirtualFileRenderer.tsx

```diff
@@ -225,9 +242,28 @@ function VirtualFileRendererComponent({
   const widthDivRef = useRef<HTMLDivElement>(null)
   const codeDisplayOverlayRef = useRef<HTMLDivElement>(null)
   const textAreaRef = useRef<HTMLTextAreaElement>(null)
+  const scrollBarRef = useRef<HTMLDivElement>(null)
   const virtualCodeRendererRef = useRef<HTMLDivElement>(null)
+  const { wrapperWidth, setWrapperRefState } = useSyncWrapperWidth()
+
+  // disable pointer events will scrolling
   useDisablePointerEvents(virtualCodeRendererRef)
-  useLeftScrollSync({ textAreaRef, overlayRef: codeDisplayOverlayRef })
+
+  // sync the width of the wrapper with the width of the text area
+  useSyncTotalWidth({ textAreaRef, widthDivRef })
+
+  // check if the code display overlay is overflowing, so we can conditionally render the scroll bar
+  const isOverflowing = useIsOverflowing(codeDisplayOverlayRef)
+
+  // sync the scroll position of the text area with the code display overlay and scroll bar
+  useSyncScrollLeft({
```

Comment:
I found the name of this hook confusing until I read the definition. Didn't know that `scrollLeft` was an element attribute. IMO `useSyncScroll` could be more clear, even though it's less specific. Nit though, your call.

Reply:
Gonna go with `useScrollLeftSync` so it reads a bit better, but I personally prefer the slightly more specific naming for clarity.